Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -93,8 +107,18 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
             driver <- SQP.findById referredDriverId >>= fromMaybeM (PersonNotFound referredDriverId.getId)
             sendNotificationToDriver driver.merchantId FCM.SHOW Nothing FCM.REFERRAL_ACTIVATED referralTitle referralMessage driver.id driver.deviceToken
           Nothing -> pure ()
-  now <- getCurrentTime
-  let rideDate = getCurrentDate now
+
+  transporterConfig <- SCT.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigNotFound merchantId.getId)
```

Comment:
Make it a single function and only call it if merchant.subscription is true in transporter config. Else it will create unnecessary entries in case of nammayatri too.